src/state/application/hooks.ts

```diff
@@ -365,7 +365,10 @@ export function useTokensPrice(tokens: (Token | NativeCurrency | null | undefine
         const key = `${token.address}_${chainId}_${version}`
         if (cache[key]) return cache[key]
 
-        const tokenPriceByETH = await getTokenPriceByETH(token?.address, client)
+        const tokenPriceByETH = await getTokenPriceByETH(
+          token?.address === ZERO_ADDRESS ? WETH[chainId as ChainId].address : token?.address,
+          client,
+        )
```

Comment:
Does `getTokenPriceByETH(WETH[chainId as ChainId].address)` always return 1?
If it does, we should simplify these line to `token?.address === ZERO_ADDRESS ? 1 : await ...`

Reply:
LOL, just a special case, we have cache for that so It's only called 1 time, I think we dont need to save 1 request, haha

Reply:
No bro, it helps much easier to understand the logic to maintain

Reply:
Make sense. moved it to condition above.